### PR TITLE
Compilation when included from C++ translation unit

### PR DESCRIPTION
**Pull request by oreyg**
Hello, I've tried including this lib from C++ translation unit, this PR includes resulting corrections...

### CODE REVIEW

**Comment by spaskalev** [comment]
Thanks! Can you please also add tests that try to compile it in C++ mode and fail if for some reason it's no longer valid cpp ?

**Comment by oreyg** [comment]
Hm, it now works, and shows (correctly) one feature of C99 that is not present in C++

**Comment by oreyg** [comment]
I'm disabling -Werr for C++, since we're more interested in whether it builds at all

**Comment by oreyg** [comment]
Sorry for spam, I don't have a configuration matching with our build-ubuntu

File: testcxx.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2020-2021 Stanislav Paskalev <spaskalev@protonmail.com>
```
Comment (spaskalev) [inline testcxx.cpp:2]: Feel free to add yourself here :) You may even want to add an AUTHORS.md file.

File: buddy_alloc.h [hunk 2]
```diff
@@ -1591,7 +1599,7 @@ static float buddy_tree_fragmentation(struct buddy_tree *t) {
         size_t pos_status = buddy_tree_status(t, state.current_pos);
         if (pos_status == 0) {
             // Empty node, process
-            size_t virtual_size = 1u << ((tree_order - state.current_pos.depth) % ((sizeof(size_t) * CHAR_BIT)-1));
+            size_t virtual_size = 1ul << ((tree_order - state.current_pos.depth) % ((sizeof(size_t) * CHAR_BIT)-1));
```
Comment (spaskalev) [inline buddy_alloc.h:1602]: Interesting, what's causing this difference between C and C++ here ? I have rather limited knowledge of cpp - might be obvious but I don't know.

**Comment by spaskalev** [comment]
> Sorry for spam, I don't have a configuration matching with our build-ubuntu

It's alright, no worries. I'm delighted to see the allocator being used and improved - some automatic emails are a small price for that :)

**Pull request merged**